x-pack/platform/plugins/shared/observability_ai_assistant/server/utils/recall/recall_and_score.ts

```diff
@@ -38,11 +43,21 @@ export async function recallAndScore({
   relevantDocuments?: RecalledSuggestion[];
   scores?: Array<{ id: string; score: number }>;
   suggestions: RecalledSuggestion[];
+  queries?: KnowledgeBaseQueryContainer[];
 }> {
-  const queries = [
-    { text: userPrompt, boost: 3 },
-    { text: context, boost: 1 },
-  ].filter((query) => query.text.trim());
+  const [[systemMessage], otherMessages] = partition(
+    messages,
+    (message) => message.message.role === MessageRole.System
+  );
```

Comment:
Personal pref nit: While `_.partition` is handy I prefer well-known functions. This is very much a nit though

```suggestion
  const systemMessage = messages.find(m => m.message.role === MessageRole.System);
  const otherMessages = messages.filter(m => m.message.role !== MessageRole.System);
```

or

```suggestion
  const isSystemMessage = m => m.message.role === MessageRole.System;
  const systemMessage = messages.find(isSystemMessage);
  const otherMessages = messages.filter(m => !isSystemMessage(m));
```